bloom/github.py

```diff
@@ -162,6 +162,15 @@ def __init__(self, username, auth, token=None):
         self.auth = auth
         self.token = token
 
+    def check_token_validity(self, username, token, update_auth=False):
+        resp = do_github_get_req('/user', self.auth)
+        resp_data = json_loads(resp)
```

Comment:
Is there any way to check the privileges the token has to see if they will be sufficient for bloom to use later?

Reply:
As far as I know the precise scopes available to a token are not visible outside the web UI. There is an authorizations API but it's not documented and according to the [Ruby library documentation](http://octokit.github.io/octokit.rb/Octokit/Client/Authorizations.html#authorizations-instance_method) it is only available using Basic username/password authentication and even with an alt that doesn't have MFA I couldn't get it to stick.